source/element_parameters.md

```diff
@@ -0,0 +1,225 @@
+# Element Parameters
+
+Lattice elements parameters are organized into **parameter groups**. 
+All groups are organized as abstract syntax trees.
+At the top level, there are the groups with names like `MagMultipole`, `ElecMultipole`, `RF`, `Alignment`, etc. 
+By convention, group names use upper camel case and it is highly recommended that this convention
+be followed but it is not mandatory.
+
+There are two element parameters that are so common that they are not grouped. 
+These element parameters are `L` (element length) and `name` (name of element).
+
+
+## Naming and Inheriting Parameter Groups
+
+Any group can be given a **name** and the values can be used in another group of the same type
+using **import**.
+For example:
+```{code} yaml
+Aperture:
+  name: ap1
+  x_limit: [-0.03, 0.03]
+```
+The above defines an aperture with the name **ap1**. 
+```{code} yaml
+Aperture:
+  name: ap2
+  inherit: ap1
+  y_limit: [-0.02, 0.02]
+```
+And the above defines a new aperture group which inherits from **ap1**.
+
+Naming a parameter group is only needed if the parameter group is defined outside of an element.
+```{code} yaml
+element:
+  name: q1
+  Aperture: 
+    x_limit: [-0.03, 0.03]
+    y_limit: [-0.02, 0.02]
+```
+And an element can inherit a parameter group from another element:
+```{code} yaml
+element:
+  name: q2
+  Aperture:
+    inherit: q1.Aperture
+```
+
+For an element to inherit all parameter groups from another element, just inherit the element itself:
+```{code} yaml
+element:
+  name: q3
+  inherit: q2
+```
+
+## Aperture Parameter Group
+
```

Comment:
As a general comment:  some of these names for components begin with the prefix `aperture_` , while others do not.  For consistency, why not use this prefix for _all_ of the components, or for _none_?  Either option would work, but we could follow the same convention for other parameter groups as well.

Reply:
Yes whatever looks best. I would use `shape` and `location` but am still partial to `aperture_shifts_with_body` (but not strongly so).

Reply:
Yes, this one is a bit special.  Either name seems fine with me.

Reply:
Done.